File: MB Website Final Bun/classes/employee.php
<?php

require_once "shift.php";
require_once "user.php";

class Employee extends User
{
    //variables
    protected $shifts;

    //constructor
    public function __construct($id, $name, $gender, $password, $email, $address, $phoneNumber, $hourlyWage, $hoursWorked, $birthDate, $holidayDays, $sickDays)
    {
        $this->shifts = array();
        parent::__construct($id, $name, $gender, $password, $email, $address, $phoneNumber, $hourlyWage, $hoursWorked, $birthDate, $holidayDays, $sickDays);
    }

    //properties
    public function GetShifts(){
        return "$this->shifts";
    }

    // public function GetEmail()
    // {
    //     parent::GetEmail();
    // }

    //methods
    public function GetShiftByStartHourAndEndHourAndDate($start, $end, $date)
    {
        $unixTimestamp = strtotime($date);
        $day = date("l", $unixTimestamp);

        foreach($shifts as &$value){
            if($value->GetStartHour() == $start && $value->GetFinishHour() == $end && $value->GetDateDay() == $day){
                return $value;
            }
        }
        return null;
    }

    public function AddShift($shift){
        array_push($this->shifts, $shift);
    }

    public function RemoveShift($shift){
        if(in_array($shift, $this->shifts)){
            unset($this->shifts[$shift]);
            return true;
        }
        else{
            return false;
        }
    }

}

File: MediaBazaar_Code_Final/MediaBazaar/classes/Employee.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MediaBazaar.classes
{
    public class Employee : User, IComparable<Employee>
    {
        //instance variables
        String department;
        private int maxMonthlyHours;

        //constructor 
        public Employee(int id, string name, string password, string email, String gender, string phoneNumber, string address, double hourlyWage, double hoursWorked, DateTime birthDate, int holidayDays, int sickDays, String department, int maxMonthlyHours) 
            : base(id, name, password, email, gender, phoneNumber, address, hourlyWage, hoursWorked, birthDate, holidayDays, sickDays)
        {
            this.department = department;
            this.maxMonthlyHours = maxMonthlyHours;
        }


        //properties
        public String Department { get { return this.department; } set { this.department = value; } }
        public int MaxMonthlyHours { get { return maxMonthlyHours; } set { maxMonthlyHours = value; }}

        public int CompareTo(Employee other)
        {
            return this.HoursWorked.CompareTo(other.HoursWorked);
        }
    }
}
